docs/design/mp_arktos.md

```diff
@@ -0,0 +1,229 @@
+```
+Operator starts creation process for Arktos network object ```vpc-1```, and then search for existing vpc object in Mizar to find vpc with id = vpc-1a2b3c4d. Once the vpc object is found, it then associated with the newly created Arktos network object ```vpc-1```: 
+    ```self.store.vpcs_arktosnet_store[vpc-1][vpc-1a2b3c4d] = vpc-1a2b3c4d object```
+
+## Pod Creation
+ Sample Pod definition:
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+  labels:
+    arktos.futurewei.com/network: vpc-1
+  annotations:
+    arktos.futurewei.com/nic: {"name": "eth0", "ip": "10.10.1.12", "subnet": "net1"}
+    arktos.futurewei.com/vip-hint: {"cidr":"10.10.1.0/26", "ip":"10.10.1.12", "policy":"BestEffort"}
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    ports:
+      - containerPort: 443
+```
+Operator kicks off creation process:
+* List Kubernetes cluster object / Arktos newtork object
+* Retrieve ```labels``` and ```annotations``` information: 
+    * ```arktos.futurewei.com/network```: vpc-1
+    * ```arktos.futurewei.com/nic```: {"name": "eth0", "ip": "10.10.1.12", "subnet": "net1"}
+* Find Arktos network object that is listed in ```arktos.futurewei.com/network```
+* Find subnet and ip information from ```arktos.futurewei.com/nic```, and choose the correct subnet to place the new pod. 
+* If neither IP nor subnet is specified, Mizar will pick the first subnet within ```vpc-1``` and assign pod to that subnet.
+* Update object store: map subnet object (which is equivalent to net object in Mizar) with pod: ```self.store.eps_net_store[net1][nginx] = pod object```
+* After Pod creation is completed, update Pod object with new annotations: ```arktos.futurewei.com/network_user_input``` and ```arktos.futurewei.com/network-readiness```.      
+* ```arktos.futurewei.com/network_user_input``` will include Mizar network information such as ```vpc name```, ```subnet name``` and ```ip address```
+* Once Pod is provisioned, operator will set ```arktos.futurewei.com/network-readiness == true```. 
+
+* In return, the annotations becomes: 
+```yaml
+  annotations:
+    arktos.futurewei.com/nic: {"name": "eth0", "ip": "10.10.1.12", "subnet": "net1"}
+    arktos.futurewei.com/vip-hint: {"cidr":"10.10.1.0/26", "ip":"10.10.1.12", "policy":"BestEffort"}
+    arktos.futurewei.com/network-readiness: "true"
+    arktos.futurewei.com/network_user_input: {"vpc": "vpc-1", "net": "net1", "ip": "10.10.1.12"}
+```
+Note: Integrity checks are needed here, where we need make sure that the ip address falls within subnet CIDR range and service CIDR range, and subnet CIDR range falls within the VPC's (vpc-1) CIDR range.
+
+
+### Service Creation 
+
+* List Kubernetes cluster object / Arktos newtork object
+* In case of using services, vpcID should not listed, and default network is selected
+* Waiting on customer to create service using yaml file:  
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-service
+spec:
+  network: my-network
```

Comment:
should use label instead. I knew this was copied from arktos dox, which has inconsistency problem.